docs/python-extensive/data/api.md

```diff
@@ -0,0 +1,469 @@
+# API
+
+<div style="text-align: center;">
+    <img src="/assets/python-extensive/data/api/api-thumbnail.png" alt="API">
+</div>
+
+An application programming interface (API) is a set of rules and protocols
+that allows one software application to interact with another.
+In other words, it is a way to communicate with a server. Some of these
+servers are openly available and host data that can be accessed by anyone.
+Others require authentication and are therefore paid services.
+
+To illustrate the practical interaction with APIs, we will retrieve
+cryptocurrency data from the [CoinCap API](https://docs.coincap.io/).
+
+We will pull a list of available cryptocurrencies, the latest price history 
+of a specific coin, plot a line chart to visualize the data and 
+lastly perform a conversion from USD to EUR.
+
+???+ info "<span style='color:red;'>Disclaimer</span>"
+    
+    This section merely demonstrates APIs on the example of cryptocurrency 
+    market data.
+
+    Cryptocurrencies involve significant financial risk. 
+    Investors should conduct thorough research and consult financial professionals 
+    before making investment decisions. The code examples presented herein are 
+    for illustrative purposes only and do not constitute financial advice 
+    nor do we endorse any companies mentioned.
+
+
+???+ question "Reading the documentation"
+    
+    Open the CoinCap API documentation ([here](https://docs.coincap.io/)) and 
+    browse through the site for a minute or two.
+
+## Rate limits
+
+During the task, you should have noticed that the API provides information 
+on rate limits. Rate limits are the number of requests that can be made to
+the server in a given time frame. If you as the user exceed the rate limit, 
+the server will respond with an error message.
+In this specific case, we can make up to 200 requests per minute which is more
+than enough for our use case. These rate limits are set by the provider and 
+can vary from one API to another. Some APIs may not have any rate limits at 
+all.
+
+But how can we even request data from the server to retrieve a list of 
+cryptocurrencies? To answer this question, the concept of endpoints is 
+introduced.
+
+## Endpoints
+
+An endpoint is a specific URL that the API uses to perform a specific action.
+For example, the endpoint `/assets` returns a list of all cryptocurrencies.
+To send a request to the server, we need to specify the endpoint in the URL.
+The server will then respond with the requested data (if everything went 
+smoothly).
+
+To request all cryptocurrencies, we can use the following URL:
+```
+https://api.coincap.io/v2/assets
+```
+
+`https://api.coincap.io/v2` is simply the URL of the API and `/assets` is 
+the endpoint of our interest.
+
+???+ question "Send your first request"
+
+    Open the URL `https://api.coincap.io/v2/assets` in your browser and 
+    observe the response.
+
+<?quiz?>
```

Comment:
Maybe we can define a new box (Admonitions) for Quiz? I think it would look a little cleaner.

Reply:
Unfortunately, with the current set-up that's not possible. The `mkdocs-quiz` plugin can't handle indentation.

Since I'm not a big fan of the syntax anyway, maybe in the future I'll write a plugin myself. From what I've seen there aren't any alternatives out there. 🙈